cmd/dataset_tokenizer/dataset_tokenizer.go

```diff
@@ -761,17 +795,32 @@ func (tt TextsTokenizer) handleExclusions(
 	return nil
 }
 
+type TokenizerStatus struct {
+	TokenizerState   string
+	PartitionerState string
+	NumTokens        int
+	PadTokens        int
+	NumFiles         int
+	CurrFile         string
+	TimeTokenizing   time.Duration
+	WaitTime         time.Duration
+	SendWait         time.Duration
+}
+
 func (tt TextsTokenizer) TokenizeTexts(
 	texts chan namedRuneReader,
 	indexPath string,
 	tokenizerPtr *gpt_bpe.GPTEncoder,
-) (chan gpt_bpe.Tokens, error) {
+) (chan gpt_bpe.Tokens, *TokenizerStatus, error) {
+	tokenizerStatus := TokenizerStatus{}
```

Comment:
`TokenizerStatus` is initialized as `tokenizerStatus` here but `status` in other places in the file.